psycop/projects/scz_bp/model_training/estimator_steps/infer_numeric_standardisation.py

```diff
@@ -0,0 +1,73 @@
+from typing import Optional
+
+import pandas as pd
+from deepchecks.utils.type_inference import infer_categorical_features
+from numpy.typing import ArrayLike
+from sklearn.preprocessing import StandardScaler
+
+from psycop.common.model_training_v2.config.baseline_registry import BaselineRegistry
+from psycop.common.model_training_v2.trainer.task.model_step import ModelStep
+
+
+class InferNumericStandardScaler(StandardScaler):
+    """Standardize features by removing the mean and scaling to unit variance.
+    This estimator automatically detects numeric columns and only standardizes them.
+    Uses deepcheck's type inference to infer numeric columns, and sklearn's StandardScaler
+    """
+
+    def __init__(
+        self,
+        *,
+        copy: bool = True,
+        with_mean: bool = True,
+        with_std: bool = True,
+    ):
+        super().__init__(
+            copy=copy,
+            with_mean=with_mean,
+            with_std=with_std,
+        )
+        self.numeric_cols: list[str] = []
+
+    def _infer_numeric_cols(self, X: pd.DataFrame) -> list[str]:
+        """Infer numeric columns from the input dataframe using deepcheck's
+        type inference. Could potentialy be simplified by interpreting colunms
+        that are integers as categorical."""
+        categorical_cols = infer_categorical_features(X)
+        numeric_cols = list(set(X.columns) - set(categorical_cols))  # type: ignore
+        return numeric_cols
+
+    def fit(  # type: ignore
+        self,
+        X: pd.DataFrame,
+        y: None = None,
+        sample_weight: ArrayLike | None = None,
+    ) -> "InferNumericStandardScaler":
```

Comment:
Could save a bit of time by type-hinting with `Self` :-)

Reply:
sfsd